### DISCUSSION TITLE
All in one card for thermostat + popup

### DISCUSSION THREAD

**Discussion by schumijo** [UI-Lovelace-Minimalist/UI#330] [Ideas]
I am working on a new AIO card for thermostat + popup card as done with light card.
I open this discussion to share ideas, improvements...

Current progress on my github : [https://github.com/schumijo/UI/tree/card_thermostat_aio_popup](https://github.com/schumijo/UI/tree/card_thermostat_aio_popup)

Actual options : 
- ulm_card_thermostat_enable_collapse : hide controls/hvac buttons when off
- ulm_card_thermostat_enable_controls: show temperature controls
- ulm_card_thermostat_enable_hvac_modes: show hvac mode buttons
- ulm_card_thermostat_enable_background_color: enable color background when heating/cooling

This card will merge : 
- card_thermostat
- custom_card_heat_pump
- custom_card_mpse_thermostat
- custom_card_httpedo13_thermostat

**Comment by edoardop13** [comment]
Good work! the only thing is the current temperature is not well highlighted

**Reply by schumijo** [reply]
@httpedo13 : I can add a variable to enable displaying the current temperature like in your custom card.

**Comment by schumijo** [comment]
Current progress

Added variables : 
- ulm_card_thermostat_enable_display_temperature
- ulm_card_thermostat_enable_horizontal

Both current temperature and target are visibles (even if they are the same on my screenshot 😅)
![Capture](https://user-images.githubusercontent.com/40740060/151556172-f558b85f-f4af-46ab-a382-f287e1d5fd4b.PNG)

**Comment by mvanlijden** [comment]
Will you automatically find the correct hvac modes or will there be an option to provide the correct ones?
I see in previous cards that Auto is missing, but Cool (which my thermostat does not support) was there.

**Reply by schumijo** [reply]
Yes. Only available modes will be displayed. And I have added auto mode
